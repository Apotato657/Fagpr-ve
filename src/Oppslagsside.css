h2 {
    margin: 1.5rem 0;
}

.ingress {
    margin-bottom: 2rem;
}

.filtrering-Wrapper{
    display: flex;
    margin-top: 4rem;
}

.filter-size {
    flex-basis: 30%;
}

.filter-Wrapper {
    margin-bottom: 24px;
}