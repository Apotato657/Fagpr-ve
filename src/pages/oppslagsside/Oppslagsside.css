h2 {
    margin: 1.5rem 0;
}

.ingress {
    margin-bottom: 2rem;
}

.result-Wrapper {
    display: flex;
    margin-top: 4rem;
    gap: 1rem;
}
.filtrering-Wrapper{
    flex-basis: 25%;
}

.filter-size {
    flex-basis: 25%;
    order:1;
}

.filter-Wrapper {
    margin-bottom: 24px;
}

.innlogging {
    margin-top: 1rem;
    display: flex;
    justify-content: flex-end;
}

.innlogging div {
    display: flex;
    align-items: center;
}

.card {
    padding: 0;
    margin: 0;
    list-style-type: none;
}

.card li {
    margin: 1rem;

}

.cardWrapper {
    color: black;
}

.pageWrapper {
    display: flex;
    margin: 2rem;
}

.paginationWrapper {
    flex-basis: 70%;
    margin: 0 2rem;
}

.pageCount {
    flex-basis: 10%;
}