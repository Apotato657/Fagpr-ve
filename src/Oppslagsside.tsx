import React from 'react';
import './Oppslagsside.css';
import {Heading, Paragraph, Search} from "@digdir/designsystemet-react";

function Oppslagsside() {
  return (
    <>
      <section>
        <Heading level={1} data-size={"xl"}>Virksomhetsopplysninger</Heading>
        <Paragraph data-size={"md"} className={'ingress'}>Her kan du gjører søk på virksomhetsnavn</Paragraph>

          <Search data-size={"md"}>
              <Search.Input aria-label='søk' placeholder={'Her kan du søke på virskomhetsnavn'}/>
              <Search.Clear/>
              <Search.Button/>
          </Search>
      </section>
    </>
  );
}

export default Oppslagsside;
