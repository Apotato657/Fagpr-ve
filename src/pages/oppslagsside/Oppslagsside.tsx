import React, {useEffect, useState} from 'react';
import './Oppslagsside.css';
import {
    Heading,
    Paragraph,
    Search, Select
} from "@digdir/designsystemet-react";
import {useAuth0} from "@auth0/auth0-react";
import {ResultCard} from "../../resultCard";
import LogoutButton from "../../Auth0/button-utlogging";
import LoginButton from "../../Auth0/button-innlogging";
import {Virksomhet, VirksomhetRespons} from "../../model/virksomhet";
import {fetchVirksomhet} from "../../virksomhet/fetch-virksomhet";
import {CustomPagination} from "../../pagination";
import {FilteringRadio, SearchFilter} from "../../filtreringPanal";

function Oppslagsside() {

    const {user, isAuthenticated} = useAuth0();
    const [virksomheter, setVirksomheter] = useState<VirksomhetRespons>()
    const [searchResult, setSearchResult] = useState<Virksomhet[]>([]);
    const [currentPage, setCurrentPage] = useState(1)
    const [searchValue, setSearchValue] = useState('')
    const [pageSize, setPageSize] = useState(3);
    const [error, setError] = useState<string>()


        const fetchVirksomhetData = async ( ) => {
            const respons = await fetchVirksomhet(searchValue, currentPage - 1, pageSize);
            if (respons.status === 'success') {
                setVirksomheter(respons.virksomheter)
            } else if (respons.status === 'fail') {
                setError(respons.error)
            }
        }

    useEffect(() => {
        fetchVirksomhetData();
    }, [pageSize, searchValue, currentPage]);


    const handelSearch = async (value: string, page = 0) => {
        if (searchValue.length > 0) {
            const fetchSearchRes = await fetchVirksomhet(value, page);
            if (fetchSearchRes.status === 'success' && fetchSearchRes.virksomheter._embedded.enheter) {
                setSearchResult(fetchSearchRes.virksomheter._embedded.enheter)
            } else if (fetchSearchRes.status === 'fail') {
                setError(fetchSearchRes.error)
            }
        }
    }

    const handelPageChange = (page:number) => {
        setCurrentPage(page);
        fetchVirksomhetData();
    }

    const handleClear = () => {
        setSearchValue('');
        setCurrentPage(1);
        setSearchResult([])
        setVirksomheter(virksomheter)
    };

    return (
        <>
            <header className='innlogging'>
                {isAuthenticated ? (
                    <div>
                        <Paragraph>Innlogget: {user?.name}</Paragraph>
                        <LogoutButton/>
                    </div>
                ) : (
                    <LoginButton/>
                )
                }
            </header>

            <section>
                <Heading level={1} data-size={"xl"}>Virksomhetsopplysninger</Heading>
                <Paragraph data-size={"md"} className={'ingress'}>Her kan du gjører søk på virksomhetsnavn</Paragraph>
                <form onSubmit={e => e.preventDefault()}>
                    <Search data-size={"md"}>
                        <Search.Input aria-label='søk'
                                      placeholder={'Her kan du søke på virskomhetsnavn'}
                                      onChange={(e) => setSearchValue((e.target.value))}
                                      value={searchValue}
                        />
                        <Search.Clear onClick={handleClear}/>
                        <Search.Button onClick={() => handelSearch(searchValue)} aria-label={'Søkeknapp'}/>
                    </Search>
                </form>
            </section>
            <div className='result-Wrapper'>
                <section className="filtrering-Wrapper">
                    <SearchFilter/>
                    <FilteringRadio/>
                </section>
                    <section>
                        <Heading
                            level={2}>Viser {virksomheter?._embedded.enheter.length} av {virksomheter?.page.totalElements} virksomheter</Heading>
                        {error &&
                            <Heading level={2}>{error}</Heading>
                        }
                        <ul className='card'>
                            {(searchResult?.length > 0 ? searchResult : virksomheter?._embedded.enheter)?.map(enhet => (
                                <li key={enhet.navn}>
                                    <ResultCard
                                        organisasjonsnummer={enhet.organisasjonsnummer}
                                        navn={enhet.navn}
                                        orgform={enhet.organisasjonsform.beskrivelse}
                                        adresse={enhet.forretningsadresse ? enhet.forretningsadresse : enhet.postadresse}
                                        adresseType={enhet.forretningsadresse ? 'Forretningsadresse' : 'Postadresse'}
                                        konkurs={enhet.konkurs}
                                    />
                                </li>
                            ))}
                        </ul>
                        {virksomheter && (
                            <div className={'pageWrapper'}>
                                <CustomPagination totalPages={virksomheter?.page?.totalPages}
                                                  sendCurrentPage={handelPageChange} getCurrentPage={currentPage}/>
                                <Select className={'pageCount'} onChange={(e) => {
                                    const newSize = parseInt(e.currentTarget.value);
                                    setPageSize(newSize);
                                    setCurrentPage(1);
                                }}>
                                    <Select.Option value={'16'}>16</Select.Option>
                                    <Select.Option value="32">32</Select.Option>
                                    <Select.Option value="64">64</Select.Option>
                                    <Select.Option value="126">126</Select.Option>
                                </Select>
                            </div>

                        )}
                    </section>
            </div>
        </>
);
}

export default Oppslagsside;
