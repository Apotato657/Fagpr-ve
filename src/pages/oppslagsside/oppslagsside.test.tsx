import {render} from "@testing-library/react";
import Oppslagsside from "./Oppslagsside";

describe('test oppslgasiden', () => {
    const setup = () => {
        const view = render(<Oppslagsside/>)
        return {view};
    }

    it('should render correctly', () => {
        const {view} = setup();


        expect(view.container).toMatchSnapshot();
    });
})