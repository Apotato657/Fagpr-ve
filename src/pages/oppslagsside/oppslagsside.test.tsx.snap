// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test oppslgasiden should render correctly 1`] = `
<div>
  <header
    class="innlogging"
  >
    <button
      class="ds-button"
      data-variant="primary"
      type="button"
    >
      Log In
    </button>
  </header>
  <section>
    <h1
      class="ds-heading"
      data-size="xl"
    >
      Virksomhetsopplysninger
    </h1>
    <p
      class="ds-paragraph ingress"
      data-size="md"
      data-variant="default"
    >
      Her kan du gjører søk på virksomhetsnavn
    </p>
    <form>
      <div
        class="ds-search"
        data-size="md"
      >
        <input
          aria-label="søk"
          class="ds-input"
          placeholder="Her kan du søke på virskomhetsnavn"
          type="search"
          value=""
        />
        <button
          aria-label="Tøm"
          class="ds-button"
          data-icon="true"
          data-variant="tertiary"
          type="reset"
        />
        <button
          aria-label="Søkeknapp"
          class="ds-button"
          data-variant="primary"
          type="submit"
        >
          Søk
        </button>
      </div>
    </form>
  </section>
  <div
    class="result-Wrapper"
  >
    <section
      class="filtrering-Wrapper"
    >
      <div
        class="filter-size"
      >
        <h2
          class="ds-heading"
          data-size="md"
        >
          Filtreringsmuligheter
        </h2>
        <div
          class="ds-field"
        >
          <label
            class="ds-label"
            data-weight="medium"
            for=":mc0geub9z6l"
          >
            Kommune
          </label>
          <u-tags
            class="ds-multi-suggestion filter-Wrapper"
            datatype="sm"
          >
            <input
              class="ds-input"
              id=":mc0geub9z6l"
              list="«r0»"
              placeholder=""
              type="text"
            />
            <button
              aria-label="Tøm"
              class="ds-button"
              data-icon="true"
              data-variant="tertiary"
              type="reset"
            />
            <u-datalist
              data-sr-plural="%d forslag"
              data-sr-singular="%d forslag"
              hidden=""
              id="«r0»"
              role="listbox"
              tabindex="-1"
            >
              <u-option
                aria-disabled="false"
                aria-selected="false"
                data-empty=""
                role="none"
                tabindex="-1"
              >
                Tomt
              </u-option>
              <u-option
                aria-disabled="false"
                aria-selected="false"
                role="option"
                tabindex="-1"
                value="Sogndal"
              >
                Sogndal
                <div>
                  Kommune
                </div>
              </u-option>
            </u-datalist>
          </u-tags>
        </div>
        <div
          class="ds-field"
        >
          <label
            class="ds-label"
            data-weight="medium"
            for=":mc0geubgfhr"
          >
            Organisasjonsform
          </label>
          <u-tags
            class="ds-multi-suggestion filter-Wrapper"
            datatype="sm"
          >
            <input
              class="ds-input"
              id=":mc0geubgfhr"
              list="«r1»"
              placeholder=""
              type="text"
            />
            <button
              aria-label="Tøm"
              class="ds-button"
              data-icon="true"
              data-variant="tertiary"
              type="reset"
            />
            <u-datalist
              data-sr-plural="%d forslag"
              data-sr-singular="%d forslag"
              hidden=""
              id="«r1»"
              role="listbox"
              tabindex="-1"
            >
              <u-option
                aria-disabled="false"
                aria-selected="false"
                data-empty=""
                role="none"
                tabindex="-1"
              >
                Tomt
              </u-option>
              <u-option
                aria-disabled="false"
                aria-selected="false"
                role="option"
                tabindex="-1"
                value="ANNA"
              >
                ANNA
                <div>
                  Annen juridisk peson
                </div>
              </u-option>
            </u-datalist>
          </u-tags>
        </div>
        <div
          class="ds-field"
        >
          <label
            class="ds-label"
            data-weight="medium"
            for=":mc0geubhaij"
          >
            Dine filter
          </label>
          <div
            class="ds-suggestion filter-Wrapper"
            datatype="md"
          >
            <input
              class="ds-input"
              id=":mc0geubhaij"
              list="«r2»"
              placeholder=""
              type="text"
            />
            <button
              aria-label="Tøm"
              class="ds-button"
              data-icon="true"
              data-variant="tertiary"
              type="reset"
            />
            <u-datalist
              data-sr-plural="%d forslag"
              data-sr-singular="%d forslag"
              hidden=""
              id="«r2»"
              role="listbox"
              tabindex="-1"
            >
              <u-option
                aria-disabled="false"
                aria-selected="false"
                data-empty=""
                role="none"
                tabindex="-1"
              >
                Tomt
              </u-option>
              <u-option
                aria-disabled="false"
                aria-selected="false"
                role="option"
                tabindex="-1"
                value="BrønnøyMix"
              >
                BrønnøyMix
              </u-option>
            </u-datalist>
          </div>
        </div>
      </div>
      <fieldset
        class="ds-fieldset filter-Wrapper"
      >
        <legend
          class="ds-label"
          data-weight="medium"
        >
          Under Konkursbehandlig
        </legend>
        <div
          class="ds-field"
        >
          <input
            class="ds-input"
            id=":mc0geubh2q5"
            type="radio"
            value="value"
          />
          <label
            class="ds-label"
            data-weight="regular"
            for=":mc0geubh2q5"
          >
            Ja
          </label>
        </div>
        <div
          class="ds-field"
        >
          <input
            class="ds-input"
            id=":mc0geubha3m"
            type="radio"
          />
          <label
            class="ds-label"
            data-weight="regular"
            for=":mc0geubha3m"
          >
            Nei
          </label>
        </div>
      </fieldset>
    </section>
    <section>
      <h2
        class="ds-heading"
      >
        Viser 
         av 
         virksomheter
      </h2>
      <ul
        class="card"
      />
    </section>
  </div>
</div>
`;
